src/lib/datatypes/include/sol-str-slice.h

```diff
@@ -123,6 +124,34 @@ sol_str_slice_to_string(const struct sol_str_slice slice)
     return strndup(slice.data, slice.len);
 }
 
+static inline struct sol_str_slice
+sol_str_slice_remove_leading_whitespace(struct sol_str_slice slice)
+{
+    char *data = (char *)slice.data;
```

Comment:
You can just do `struct sol_str_slice copy = slice;`, and then operate directly on `copy`.